src/cata_tiles.cpp

```diff
@@ -1572,27 +1573,66 @@ void cata_tiles::draw( const point &dest, const tripoint &center, int width, int
-            draw_terrain( pos, ll, height_3d, invisible );
-
             draw_points.emplace_back( pos, height_3d, ll, invisible );
         }
-        const std::array<decltype( &cata_tiles::draw_furniture ), 13> drawing_layers = {{
-                &cata_tiles::draw_furniture, &cata_tiles::draw_graffiti, &cata_tiles::draw_trap, &cata_tiles::draw_part_con,
+
+        // List all layers for a single z-level
+        const std::array<decltype( &cata_tiles::draw_furniture ), 11> drawing_layers = {{
+                &cata_tiles::draw_terrain, &cata_tiles::draw_furniture, &cata_tiles::draw_graffiti, &cata_tiles::draw_trap, &cata_tiles::draw_part_con,
+                &cata_tiles::draw_field_or_item,
+                &cata_tiles::draw_vpart_no_roof, &cata_tiles::draw_vpart_roof,
+                &cata_tiles::draw_critter_at, &cata_tiles::draw_zone_mark,
+                &cata_tiles::draw_zombie_revival_indicators
+            }
+        };
+        // Legacy code for isometric tilesets until they are ready
+        const std::array<decltype( &cata_tiles::draw_furniture ), 14> drawing_layers_legacy = {{
+                &cata_tiles::draw_terrain, &cata_tiles::draw_furniture, &cata_tiles::draw_graffiti, &cata_tiles::draw_trap, &cata_tiles::draw_part_con,
                 &cata_tiles::draw_field_or_item, &cata_tiles::draw_vpart_below,
                 &cata_tiles::draw_critter_at_below, &cata_tiles::draw_terrain_below,
                 &cata_tiles::draw_vpart_no_roof, &cata_tiles::draw_vpart_roof,
                 &cata_tiles::draw_critter_at, &cata_tiles::draw_zone_mark,
                 &cata_tiles::draw_zombie_revival_indicators
             }
         };
-        // for each of the drawing layers in order, back to front ...
-        for( auto f : drawing_layers ) {
-            // ... draw all the points we drew terrain for, in the same order
+        if( is_isometric() ) {
+            // Disable multi z-level display on isometric tilesets. Can be enabled but will cause the following issues:
+            // 1. Supposedly transparent terrain and fields such as t_open_air need to be given sprites
+            // Otherwise they will be replaced with an opaque fallback sprite
+            // 2. May worsen visibility issues in isometric tilesets with inconsistent terrain heights
+            for( tile_render_info &p : draw_points ) {
+                for( auto f : drawing_layers_legacy ) {
+                    ( this->*f )( p.pos, p.ll, p.height_3d, p.invisible );
+                }
+            }
+        } else {
             for( tile_render_info &p : draw_points ) {
-                ( this->*f )( p.pos, p.ll, p.height_3d, p.invisible );
+                // Find lowest z-level to draw
+                tripoint p_draw = p.pos;
+                int cur_height_3d = p.height_3d;
+                while( !here.dont_draw_lower_floor( p_draw ) && p_draw.z > -10 ) {
```

Comment:
The same, but without magic number.
```suggestion
                while( !here.dont_draw_lower_floor( p_draw ) && p_draw.z > -OVERMAP_DEPTH) {
```